Modules/_cursesmodule.c

```diff
@@ -184,25 +194,25 @@ static const char *curses_screen_encoding = NULL;
 #define PyCursesSetupTermCalled                                         \
     do {                                                                \
         if (curses_setupterm_called != TRUE) {                          \
-            PyErr_SetString(PyCursesError,                              \
+            PyErr_SetString(curses_global_state.PyCursesError,          \
```

Comment:
Please don't refer to curses_global_state since it's going to be removed in following commits, but call get_cursesmodule_state(module) instead. Or even pass the state to the macro?

Same remark below.

Reply:
Unfortunately the state cannot be used in all contexts :( some functions are parameterless... that's why for some functions I will need to get the error type using an import.

Reply:
(Only the functions from the C API will need this kind of work. So I'll need an additional macro to handle the case where I have access to the module's state and when I don't).

Reply:
So I managed to more modifications. Do you think it's fine like this for now? (the trick with the import will be needed later so I'll leave it already for now).